pkg/controllers/awsloadbalancercontroller/rbac_clusterrole.go

```diff
@@ -2,107 +2,29 @@ package awsloadbalancercontroller
-
-	albo "github.com/openshift/aws-load-balancer-operator/api/v1alpha1"
 )
 
-// ensureClusterRole creates and maintains a clusterrole required by the aws-load-balancer controller.
-func (r *AWSLoadBalancerControllerReconciler) ensureClusterRole(ctx context.Context, controller *albo.AWSLoadBalancerController) error {
-	reqLogger := log.FromContext(ctx)
-
-	desired := desiredClusterRole(ctx, fmt.Sprintf("%s-%s", controllerResourcePrefix, controller.Name))
-
-	reqLogger.Info("ensuring clusterroles", "clusterroles", desired.Name)
-
-	if err := controllerutil.SetControllerReference(controller, desired, r.Scheme); err != nil {
-		return fmt.Errorf("failed to set the controller reference for clusterroles %s : %w", desired.Name, err)
-	}
+// verifyClusterRole verifies whether the given cluster role exists.
+func (r *AWSLoadBalancerControllerReconciler) verifyClusterRole(ctx context.Context, name string) (bool, error) {
+	log.FromContext(ctx).Info("verifying clusterrole", "name", name)
 
-	exist, current, err := r.currentClusterRole(ctx, desired)
-	if err != nil {
-		return fmt.Errorf("failed to fetch current clusterroles: %w", err)
-	}
-
-	if !exist {
-		if err := r.createClusterRole(ctx, desired); err != nil {
-			return err
-		}
-		_, current, err = r.currentClusterRole(ctx, desired)
-		if err != nil {
-			return fmt.Errorf("failed to fetch created clusterroles: %w", err)
-		}
-		reqLogger.Info("created clusterroles", "clusterroles", desired.Name)
-	}
-
-	if err := r.updateClusterRole(ctx, current, desired); err != nil {
-		return fmt.Errorf("failed to update clusterroles: %w", err)
-	}
-
-	return nil
+	exists, _, err := r.currentClusterRole(ctx, name)
+	return exists, err
 }
 
-// updateClusterRoles updates the current clusterroles and returns a flag to denote if the update was done.
-func (r *AWSLoadBalancerControllerReconciler) updateClusterRole(ctx context.Context, current *rbacv1.ClusterRole, desired *rbacv1.ClusterRole) error {
-	changed := hasClusterRoleChanged(current, desired)
-
-	if !changed {
-		return nil
-	}
-
-	updated := current.DeepCopy()
-	updated.Rules = desired.Rules
-
-	if err := r.Client.Update(ctx, updated); err != nil {
-		return err
-	}
-
-	return nil
-}
-
-func hasClusterRoleChanged(current *rbacv1.ClusterRole, desired *rbacv1.ClusterRole) bool {
-	return !(reflect.DeepEqual(current.Rules, desired.Rules))
-}
-
-func (r *AWSLoadBalancerControllerReconciler) createClusterRole(ctx context.Context, role *rbacv1.ClusterRole) error {
-	if err := r.Client.Create(ctx, role); err != nil {
-		if !errors.IsAlreadyExists(err) {
-			return fmt.Errorf("failed to create clusterrole %s/%s: %w", role.Namespace, role.Name, err)
-		}
-	}
-
-	return nil
-}
-
-func (r *AWSLoadBalancerControllerReconciler) currentClusterRole(ctx context.Context, role *rbacv1.ClusterRole) (bool, *rbacv1.ClusterRole, error) {
+func (r *AWSLoadBalancerControllerReconciler) currentClusterRole(ctx context.Context, name string) (bool, *rbacv1.ClusterRole, error) {
```

Comment:
Do we really need 2 different functions: `currentClusterRole` and `verifyClusterRole`. Couldn't the `currentClusterRole` be used directly in the reconcile function?

Reply:
Kept the single method